<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>945</width>
    <height>811</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>945</width>
    <height>811</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Master Dark Frame Maker</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0" rowspan="4">
     <widget class="QTableView" name="filesTable">
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="tabKeyNavigation">
       <bool>false</bool>
      </property>
      <property name="showDropIndicator" stdset="0">
       <bool>false</bool>
      </property>
      <property name="dragDropOverwriteMode">
       <bool>false</bool>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <property name="sortingEnabled">
       <bool>true</bool>
      </property>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <attribute name="verticalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
     </widget>
    </item>
    <item row="0" column="1" colspan="2">
     <widget class="QFrame" name="frame_2">
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>200</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>9999</width>
        <height>9999</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="verticalSpacing">
        <number>8</number>
       </property>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Pre-Combine Bias File Calibration&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QRadioButton" name="noPreClalibrationRB">
         <property name="toolTip">
          <string>No precalibration</string>
         </property>
         <property name="text">
          <string>No pre-calibration</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">preCalibrationGroup</string>
         </attribute>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QRadioButton" name="fixedPedestalRB">
         <property name="toolTip">
          <string>Subtract a given fixed pedestal amount from each pixel</string>
         </property>
         <property name="text">
          <string>Pedestal</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">preCalibrationGroup</string>
         </attribute>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="fixedPedestalAmount"/>
       </item>
       <item row="3" column="0">
        <widget class="QRadioButton" name="autoPreCalibrationRB">
         <property name="toolTip">
          <string>Automatically choose a matching Bias file from a directory of Bias files</string>
         </property>
         <property name="text">
          <string>Auto from Directory</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">preCalibrationGroup</string>
         </attribute>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="setAutoDirectory">
         <property name="text">
          <string>Set Dir</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="autoDirectoryName">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Directory name echoed here&lt;br/&gt;second line&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QRadioButton" name="fixedPreCalFileRB">
         <property name="toolTip">
          <string>Use fixed bias file specified here for precalibration</string>
         </property>
         <property name="text">
          <string>Fixed Bias File:</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">preCalibrationGroup</string>
         </attribute>
        </widget>
       </item>
       <item row="5" column="1" rowspan="2">
        <widget class="QPushButton" name="selectPreCalFile">
         <property name="text">
          <string>Select File</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="precalibrationPathDisplay">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;File name is echoed here&lt;br/&gt;Second line&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1" colspan="2">
     <widget class="QFrame" name="frame_3">
      <property name="minimumSize">
       <size>
        <width>311</width>
        <height>177</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>311</width>
        <height>177</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Combination Algorithm&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QRadioButton" name="combineMeanRB">
         <property name="toolTip">
          <string>Simple mean. Best signal-to-noise, but outliers (satellites etc) will show up.</string>
         </property>
         <property name="text">
          <string>Mean</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="message">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QRadioButton" name="combineMedianRB">
         <property name="toolTip">
          <string>Simple median. Rejects outliers, but signal-to-noise not as good as with Mean.</string>
         </property>
         <property name="text">
          <string>Median</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QRadioButton" name="combineMinMaxRB">
         <property name="toolTip">
          <string>Drop min and max values, then Mean. Works well when outliers are hot or black pixels.</string>
         </property>
         <property name="text">
          <string>Min/Max Clip</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Drop</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLineEdit" name="minMaxNumDropped">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;How many min and max values are dropped?  (1 = drop all instances of the max and all instances of the min; 2 = all instances of max and next-to-max and all instances of 2 min values, etc.)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QRadioButton" name="combineSigmaRB">
         <property name="toolTip">
          <string>Drop pixels more than a given sigma ratio from the mean, then mean the rest. Good rejection of anomalous data.</string>
         </property>
         <property name="text">
          <string>Sigma Clip</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Z Limit</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QLineEdit" name="sigmaThreshold">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sigma ratio (how many standard deviations the value is from the mean) beyond which data are dropped.  Try 2.0 then experiment.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="3">
        <widget class="QCheckBox" name="ignoreFileType">
         <property name="text">
          <string>Ignore FITS file type, allow combine</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1" colspan="2">
     <widget class="QFrame" name="frame_4">
      <property name="minimumSize">
       <size>
        <width>311</width>
        <height>125</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>311</width>
        <height>125</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0" colspan="3">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Process Multiple Files at Once, Grouped&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QCheckBox" name="groupBySizeCB">
         <property name="toolTip">
          <string>Process sets of files having different dimensions or binnings</string>
         </property>
         <property name="text">
          <string>Group by size (dimensions and binning)</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="groupByExposureCB">
         <property name="toolTip">
          <string>Process sets of files having different exposure times</string>
         </property>
         <property name="text">
          <string>Group by Exposure, within:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="exposureGroupTolerance">
         <property name="maximumSize">
          <size>
           <width>31</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Consider exposure times to be the same if the are within this percentage of each other.</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>%</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QCheckBox" name="groupByTemperatureCB">
         <property name="toolTip">
          <string>Process sets of files having different CCD temperatures</string>
         </property>
         <property name="text">
          <string>Group by Temperature, within:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="temperatureGroupTolerance">
         <property name="maximumSize">
          <size>
           <width>31</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Consider temperatures to be the same if the are within this percentage of each other.</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>%</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="1" colspan="2">
     <widget class="QFrame" name="frame_5">
      <property name="minimumSize">
       <size>
        <width>311</width>
        <height>101</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>311</width>
        <height>101</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Disposition of Input Files &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QRadioButton" name="dispositionNothingRB">
         <property name="toolTip">
          <string>Leave the input files alone.</string>
         </property>
         <property name="text">
          <string>Do Nothing</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">dispositionGroup</string>
         </attribute>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QRadioButton" name="dispositionSubFolderRB">
         <property name="toolTip">
          <string>Move the input files to a sub-folder (in the same directory as the input files) with the given name</string>
         </property>
         <property name="text">
          <string>Move to:</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">dispositionGroup</string>
         </attribute>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="subFolderName">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Name for the folder to move the processed files. Optionally this name can contain %d, %t, or %f, which will be replaced by the date, time, and filter name, respectively.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="4" column="0" colspan="2">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="selectNoneButton">
        <property name="text">
         <string>Select None</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="selectAllButton">
        <property name="text">
         <string>Select All</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="selectedLabel">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>538</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="5" column="0" colspan="2">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="pickFilesButton">
        <property name="text">
         <string>Pick Files</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>518</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="5" column="2">
     <widget class="QPushButton" name="combineSelectedButton">
      <property name="text">
       <string>Combine Selected</string>
      </property>
      <property name="autoDefault">
       <bool>true</bool>
      </property>
      <property name="default">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>945</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionSelectAll"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+;</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="preCalibrationGroup"/>
  <buttongroup name="dispositionGroup"/>
 </buttongroups>
</ui>
